documentation/guides/migrating-from-v4-to-v5.md

```diff
@@ -0,0 +1,155 @@
+
+  return (
+    <AppProvider apiKey={config.apiKey} shopOrigin={config.shopOrigin}>
+      <Modal /* ... *//>
+    </AppProvider>
+  );
+}
+
+
+// New
+import {AppProvider} from '@shopify/polaris';
+import {Provider, TitleBar} from '@shopify/app-bridge-react';
+
+function MyApp() {
+  const config = {apiKey: '12345', shopOrigin: shopOrigin};
+
+  return (
+    <AppProvider>
+      <Provider config={config}>
+        <Modal /* ... *//>
+      </Provider>
+    <AppProvider>
+  );
+}
+```
+
+## AppProvider's i18n translations array is now preferred language first
+
+AppProvider's i18n prop accepts an array of translation dictionaries so that if a translation key in one language is not found it can try a fallback language.
+
+In v4 this order was `[fallbackDictionary, preferredDictionary]`. This was unintuitive, and does not align with other translation tools used at Shopify.
+In v5 this order is `[preferredDictionary, fallbackDictionary]`.
+
+If you are using `react-i18n` to provide async translations you can now pass translations in the order returned by `react-i18n`:
+
+```diff
+import AppProvider from '@shopify/polaris';
+// en.json is English. Replace with fr.json for French, etc
+import translations from '@shopify/polaris/locales/en.json';
+import {useI18n} from '@shopify/react-i18n';
+
+function App() {
+  const [i18n] = useI18n({
+    id: 'Polaris',
+    fallback: translations,
+    translations(locale) {
+      return import(
+        /* webpackChunkName: "Polaris-i18n", webpackMode: "lazy-once" */ `@shopify/polaris/locales/${locale}.json`
+      ).then((dictionary) => dictionary && dictionary.default);
+    },
+  });
+
+  return (
+-    <AppProvider i18n={i18n.translations.reverse()}>{/* App content */}</AppProvider>
++    <AppProvider i18n={i18n.translations}>{/* App content */}</AppProvider>
+  );
+}
+```
+
+## Build output changes - general consumers
+
+Our compiled styles file has moved. Imports must be updated:
+
+```diff
+- import '@shopify/polaris/styles.css';
++ import '@shopify/polaris/dist/styles.css';
+```
+
+Our minified styles file has been removed, and you should use the above file instead.
+
+```diff
+- import '@shopify/polaris/styles.min.css';
++ import '@shopify/polaris/dist/styles.css';
+```
+
+Our scss entrypoint has been removed, and you should use the above styles file instead. If you require our mixins/functions you should also import the our public-api entrypoint.
+
+```diff
+- @import 'path_to_node_modules/@shopify/polaris/styles';
++ @import 'path_to_node_modules/@shopify/polaris/dist/styles.css';
++ /* Optional - only needed if you use our mixins/functions */
++ @import 'path_to_node_modules/@shopify/polaris/dist/styles/public-api';
+
+## Build output changes - @shopify/sewing-kit integrations only
+
+Apps using `@shopify/sewing-kit` - Shopify's in-house opinionated build toolkit - will need to update to at least v0.132.2 to make use of the reworked "esnext" build output in polaris v5.
```

Comment:
I don't think "Shopify's in-house opinionated build toolkit" adds value here.

```suggestion
Apps using `@shopify/sewing-kit` will need to update to at least v0.132.2 to make use of the reworked "esnext" build output in polaris v5.
```

Reply:
I was trying to enforce the idea that this doesn't apply to partners, and thus they can skip over this section. By adding this little bit I was trying to preempt thoughts of "what's sewing-kit? does this apply to me?" that may occur we just refer to sewing-kit by name.

Do you think just referring to the package by name is enough for partners to know this doesn't apply to them, is there there some other way we can help them reach that "cool, this doesn't apply to me" realization quickly?

Reply:
I think referring to the package is enough. Happy for other opinions though.

Reply:
I'm with Ben on this one, unless you feel strongly about removing this part of the sentence 👍